pandas/core/internals.py

```diff
@@ -1898,6 +1897,37 @@ def is_view(self):
         """Extension arrays are never treated as views."""
         return False
 
+    def setitem(self, indexer, value, mgr=None):
+        """Set the value inplace, returning a same-typed block.
+
+        This differs from Block.setitem by not allowing setitem to change
+        the dtype of the Block.
+
+        Parameters
+        ----------
+        indexer : tuple, list-like, array-like, slice
+            The subset of self.values to set
+        value : object
+            The value being set
+        mgr : BlockPlacement
+
+        Returns
+        -------
+        Block
+
+        Notes
+        -----
+        'indexer' is a direct slice/positional indexer. 'value' must
+        be a compatible shape.
+        """
+        if isinstance(indexer, tuple):
```

Comment:
hmm, you are doing to for compatibility? this is not very robust and should be handled by the caller. indexing is quite tricky as the contracts for who converts what are almost all handled in core/indexing.py So the internals are so of you-know-what-you-are doing. Since you are exposing this almost directly, I think you need some handling routines (and am talking about the indexers here) IN EA, which actually call things in core/indexing.py.

Reply:
> hmm, you are doing to for compatibility?

I guess, in the sense that things didn't work without it :) I think at some point, probably in `indexing.py`, the indexer is translated to a 2-D indexer, even though we know that EAs are always 1-D.

> Since you are exposing this almost directly

I don't think the call stack is any different for setting on an ExtensionBlock vs. anything else.

> I think you need some handling routines (and am talking about the indexers here) IN EA

IIUC, you're saying that the translation from a 2-D indexer to a 1-D indexer should happen elsewhere, yes? And the validation too? That sounds reasonable.

Reply:
@jreback what place in `core/indexing.py` did you have in mind? `_setitem_with_indexer`?

I don't see anywhere there that deals with specific blocks, just whether or not the block managner is holding mixed types (split path or not).